localgov.install

```diff
@@ -31,8 +30,8 @@ function localgov_update_8201() {
   if ($module_handler->moduleExists('search_api')) {
     $manager = \Drupal::entityDefinitionUpdateManager();
     $entity_type = $manager->getEntityType('search_api_task');
-    if ($entity_type->getHandlerClass('storage_schema') != TaskStorageSchema::class) {
-      $entity_type->setHandlerClass('storage_schema', TaskStorageSchema::class);
+    if ($entity_type->getHandlerClass('storage_schema') != 'TaskStorageSchema') {
```

Comment:
Hi Steve, the search_api module's [search_api_update_8110()](https://git.drupalcode.org/project/search_api/-/blob/e8bb207776db70b8a35768ad3b2a59bbe31d7149/search_api.install#L406) has taken a slightly different approach.  I don't know if that's relevant here but thought I will mention it anyway. @stephen-cox

Reply:
Hi, sorry had seen this on.

Adnan well spotted that.

The full patch https://git.drupalcode.org/project/search_api/-/commit/e8bb207776db70b8a35768ad3b2a59bbe31d7149 deprecates the TaskStorageSchema class completely.

Reply:
Happy for someone to suggest the changes required here; I'm flat out and not likely to have time to look at this for a few days and tests are failing until this gets fixed.

Reply:
Having deprecated code in a install file seems fine to me. Like it will need to be removed and a major version update made when the code goes. But this is the same update that search api itself also still has https://git.drupalcode.org/project/search_api/-/blob/e8bb207776db70b8a35768ad3b2a59bbe31d7149/search_api.install#L341 we needed to run it because of that blunder move of patching an update.

Search API will then in its updates do it's next update https://git.drupalcode.org/project/search_api/-/blob/e8bb207776db70b8a35768ad3b2a59bbe31d7149/search_api.install#L406

Or is the problem that localgov updates are run _after_ search api's on installs that haven't yet run localgov_update_8201(). So localgov_update_8201() undoes search_api_update_8110()? If that's the case... then we should rewrite history of the updates; swapping it for the same as search_api_update_8110() would seem 'harmless' (in the I don't think it digs us further into the hole).

Reply:
So just on the hole digging thing. To run it past everyone:-

Some people will have run localgov_update_8201() and should have run some search_api updates < 8110 at which point they will run 8110 with the next update - no problem for them.
Other people won't have run localgov_update_8201 at which point they'll also be either search_api updates < 8107, or search_api_updates with whatever we patched as 8107. Running a localgov_update_8201 and also running all the search api updates from either before 8107 or from 8108 is _safe_?
Can we get a couple of agreements on that last sentence. Then we can change it?